frontend/canvas-chrome-ext/src/scripts/instructor-inject-script.jsx

```diff
@@ -114,19 +114,62 @@ function getParameters() {
 function generateReadOnlyCodeView(submissionFiles, instructorViewContainer) {
     document.getElementById("iframe_holder").style.display = "none";
 
+    console.log("generate terminal view function");
+    let appBar = InstructorView_appBar();
     let tabContainer = initTabContainer();
     let codeContainer = initCodeContainer();
-    let darkModeButton = initDarkModeButton();
-
-    instructorViewContainer.appendChild(darkModeButton);
-    instructorViewContainer.appendChild(tabContainer);
-    instructorViewContainer.appendChild(codeContainer);
 
     let isInDarkMode = false;
     let fileName = submissionFiles[0].name;
     let previousCodeWindowId = getCodeWindowId(fileName);
     let previousTabId = getTabId(fileName)
 
+
+    let darkModeButton = initDarkModeButton();
+    let playButton = initPlayButton();
+    let abortButton = initAbortButton();
+
+    //Darkmode toggle
+    darkModeButton.addEventListener("click", function () {
+        let codeContainerChildElements = codeContainer.getElementsByTagName("textarea");
+        for (var i = 0; i < codeContainerChildElements.length; i++) {
+            if (isInDarkMode) {
+                let textAreaElement = codeContainerChildElements[i];
+                textAreaElement.style.backgroundColor = "#f1f1f1";
+                textAreaElement.style.color = "black"
+            } else {
+                let textAreaElement = codeContainerChildElements[i];
+                textAreaElement.style.backgroundColor = "black";
+                textAreaElement.style.color = "white";
+            }
+        }
+        isInDarkMode = !isInDarkMode; // Flip to other mode
+    });
+
+    //Play button listener
+    playButton.addEventListener("click", async function (){
+        //send post request to backend to start compilation or send makefile?
+        console.log('calling compilation from instructor view');
+        setTimeout(async () => {
```

Comment:
5 sec timeout when making the API call is probably not necessary.  Would make more sense to wrap the entire playButton event listener in the terminal frame `load` listener because we would only want to execute the code once the terminal frame has completely loaded. Once it's loaded, the API call can be made right away, so no need for a timeout